server.js

```diff
@@ -23,23 +24,21 @@ const port = process.env.PORT || 3000
 
 app.prepare().then(() => {
   createServer((req, res) => {
-    const parsedUrl = parse(req.url, true)
-    const { pathname, query } = parsedUrl
+    const parsedUrl = parseURL(req.url)
+    const { pathname, queryStr } = parsedUrl
     const host = req.headers.host
 
-    /*
-     * HTTP redirects
-     */
     let [redirectCode, redirectLocation] = getRedirect(host, pathname, {
       req,
       dev
     })
-
     if (redirectLocation) {
-      // should be getting the query as a string
-      const { query } = parse(req.url)
-      if (query) {
-        redirectLocation += '?' + query
+      /*
+       * HTTP redirects
+       */
+
+      if (queryStr) {
+        redirectLocation += '?' + queryStr
```

Comment:
No need to call `url.parse` twice because we can `querystring.parse` the query string instead.
Same result